util/collate/collate.go

```diff
@@ -35,17 +35,44 @@ type CollatorOption struct {
 type Collator interface {
 	// Compare returns an integer comparing the two strings. The result will be 0 if a == b, -1 if a < b, and +1 if a > b.
 	Compare(a, b string, opt CollatorOption) int
-	// Key returns the collate key for str.
+	// Key returns the collate key for str. If the collation is padding, make sure the PadLen >= len(rune[]str) in opt.
 	Key(str string, opt CollatorOption) []byte
 }
 
 // SetNewCollationEnabled sets if the new collation are enabled.
 func SetNewCollationEnabled(flag bool) {
 	setCollationOnce.Do(func() {
-		newCollationEnabled = flag
+		SetNewCollationEnabledForTest(flag)
 	})
 }
 
+// SetNewCollationEnabledForTest sets if the new collation are enabled in test.
+// Note: Be careful to use this function, if this functions is used in tests, make sure the tests are serial.
+func SetNewCollationEnabledForTest(flag bool) {
+	newCollationEnabled = flag
+	if newCollationEnabled {
+		collatorMap["utf8mb4_bin"] = &binPaddingCollator{}
+		collatorMap["utf8_bin"] = &binPaddingCollator{}
+		collatorMap["utf8mb4_general_ci"] = &generalCICollator{}
+		collatorMap["utf8_general_ci"] = &generalCICollator{}
+
+		collatorIDMap[46] = &binPaddingCollator{}
```

Comment:
I suggest that we define them in protobuf

Reply:
Prefer to rename `binPaddingCollator` to `binCollator` and rename original `binCollator` to `binNotPaddingCollator`.

Reply:
> Defining constants is better than using magic numbers.

It need to update parser. And for the deadline reason, we can only use the magic number.

Reply:
> Prefer to rename `binPaddingCollator` to `binCollator` and rename original the `binCollator` to `binNotPaddingCollator`.

Ok, it will be updated in future.